backend/corpora/common/entities/collection.py

```diff
@@ -260,7 +260,7 @@ def publish(self, data_submission_policy_version):
             )
             for link in self.links:
                 CollectionLink(link).update(collection_id=self.revision_of, commit=False)
-            public_collection.links = self.links
+            self.session.expire(public_collection)
```

Comment:
@Bento007 you were right! I thought this code was correct initially for a different reason, but yeah this was overwriting the public collection's links with soon-to-be-deleted entities.

Reply:
Technically we don't, as the code currently stands. I did this to prevent mistakes from being made in the future, in case someone adds further link logic in this method: with all other entity modifications (such as the `update` call), updates will be available immediately, but with the link reassignment, those won't be available unless you `expire` this object. So trying to keep the behavior consistent.

Reply:
If you expire the object without committing the changes will the changes persist?

Reply:
I suggest removing it.

Reply:
> If you expire the object without committing will the changes persist?

Yes, the `expire` just tells sqlalchemy to refetch the object (and its relations) in all subsequent usages of `public_collection`.